cmd/crossplane/main.go

```diff
@@ -17,22 +17,43 @@ limitations under the License.
 package main
 
 import (
-	"log"
+	"os"
+	"path/filepath"
 	"time"
 
+	"gopkg.in/alecthomas/kingpin.v2"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	runtimelog "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
 
 	"github.com/crossplaneio/crossplane/pkg/apis"
 	"github.com/crossplaneio/crossplane/pkg/controller"
+	"github.com/crossplaneio/crossplane/pkg/log"
 )
 
 func main() {
+	var (
+		logger = log.Log
+
+		app   = kingpin.New(filepath.Base(os.Args[0]), "An open source multicloud control plane.").DefaultEnvars()
```

Comment:
@jbw976 I'm not aware of any command suggestion support in Kingpin.

I picked Kingpin in this instance because I wanted to add support for a single flag without spending too much time on it. Kingpin is what I know, and it has a very succinct and simple API that I could easily implement. I personally find Cobra's API to be confusing and overcomplicated for simple CLI apps, which has disincentivised me from learning it.

I don't feel strongly about what CLI interface we use. Perhaps someone would like to switch us over to Cobra in a future PR - I believe (apart from env var handling?) this should be possible without breaking the 'API' presented to users by CLI flags.

Reply:
I think so too @negz, I am not married to Cobra and I think using Kingpin as your have described is OK with me.